### PR TITLE
SRE-417: Test using Docker Compose

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

File: celery-entrypoint.sh [hunk 1]
```diff
@@ -30,13 +30,11 @@ if [ "$1" = 'celery' ]; then
...
-  # Set the schedule file if this is beat
-  if [ "$2" = 'beat' ]; then
-    set -- "$@" --schedule /var/run/celery/celerybeat-schedule
-  fi
-
   # Run under the celery user
   set -- su-exec celery "$@"
+
+  # Change to the Celery working directory (only place the user can write)
+  cd /var/run/celery
```
Comment (JayH5) [inline celery-entrypoint.sh:37]: This is the actual bug that needed to be fixed 😅... Celery Beat creates a file `celerybeat.pid` by default in the current working directory..which it didn't have permissions to write to. We now just change directory to the Celery working directory and let Celery Beat create its schedule and PID file in its default positions.

File: example/mysite/settings.py [hunk 2]
```diff
@@ -23,9 +23,9 @@
 SECRET_KEY = '2^!@63mf-#%l8m_v*o3hz(&2iw35o_6h^3^&x$+1rm558^)_i9'
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = False
+DEBUG = True
 
-ALLOWED_HOSTS = ['127.0.0.1', 'localhost']
+ALLOWED_HOSTS = []
```
Comment (JayH5) [inline example/mysite/settings.py:28]: These changes are me just setting stuff back to the default created by `django-admin startproject`.. these are overridden in the Docker settings.

File: example/mysite/docker_settings.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
```
Comment (JayH5) [inline example/mysite/docker_settings.py:1]: This whole file could do with some documentation, but I am planning to do that in a separate PR where I update all the documentation for the project and use this settings file as an example of some best practices.

File: example/test.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+# Set a trap on errors to make it clear when tests have failed
+trap '{ set +x; echo; echo FAILED; echo; } >&2' ERR
+
+set -x
+
+# Bring up the DB and AMQP first
+docker-compose up -d db amqp && sleep 5
+
+# Bring everything else up
+docker-compose up -d && sleep 5
```
Comment (smn) [inline example/test.sh]: The way I've gone about this is to using a combination of `timeout` and grepping: https://github.com/rapidpro/rapidpro-docker/blob/master/.travis.yml#L45, not sure what you'd prefer. If `sleep 5` is enough to not cause random build failures then let's keep it as its simpler.

**Comment by smn** [comment]
:+1: from me
